covizu/clustering.py

```diff
@@ -31,38 +32,39 @@ def recode_features(records, callback=None, limit=10000):
     # compress genomes with identical feature vectors
     fvecs = {}
     for muts, variant in records.items():
-        key = tuple([tuple(x.split('|')) for x in muts.split(',')])
+        key = tuple(tuple(x.split('|')) for x in muts.split(','))
         if key not in fvecs:
             fvecs.update({key: []})
         for sample in variant:
-            label = "{covv_virus_name}|{covv_location}|{covv_accession_id}|{covv_collection_date}".format(**sample)
+            label = (f"{sample['covv_virus_name']}|{sample['covv_location']}|"
+                    f"{sample['covv_accession_id']}|{sample['covv_collection_date']}")
+
             fvecs[key].append(label)
 
     # limit to N most recently-sampled feature vectors
-    intermed = [(max([label.split('|')[-1] for label in labels]), key)
+    intermed = [(max(label.split('|')[-1] for label in labels), key)
                 for key, labels in fvecs.items()]
     intermed.sort(reverse=True)
 
     # generate union of all features
     if callback:
-        callback("Reduced to {} variants; generating feature set union".format(len(fvecs)))
-    union = {}
-    labels = {}
-    indexed = []
+        callback(
+            f"Reduced to {len(fvecs)} variants; generating feature set union")
+    intermed_out = [{}, {}, []] # [{union}, {labels}, [idx]]
```

Comment:
I did this to lower the amount of local variables, but I feel like it hurts readability more than anything